shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -36,97 +36,88 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
 
         public static string GetTypeDisplayName(Type type, bool fullName = true)
         {
-            var sb = new StringBuilder();
-            ProcessTypeName(type, sb, fullName);
-            return sb.ToString();
+            var builder = new StringBuilder();
+            ProcessTypeName(builder, type, fullName);
+            return builder.ToString();
         }
 
-        private static void AppendGenericArguments(Type[] args, int startIndex, int numberOfArgsToAppend, StringBuilder sb, bool fullName)
+        private static void ProcessTypeName(StringBuilder builder, Type type, bool fullName)
         {
-            var totalArgs = args.Length;
-            if (totalArgs >= startIndex + numberOfArgsToAppend)
+            if (type.GetTypeInfo().IsGenericType)
```

Comment:
Shouldn't need `GetTypeInfo()`. This will also remove this `using System.Reflection;`